integration-tests/test-oncotree-dependent-tools.sh

```diff
@@ -0,0 +1,124 @@
+#!/bin/bash
+
+TESTING_DIRECTORY=/var/lib/jenkins/tempdir
+if [ ! -d $TESTING_DIRECTORY ] ; then
+    mkdir -p $TESTING_DIRECTORY
+fi
+TESTING_DIRECTORY_TEMP=$(mktemp -d $TESTING_DIRECTORY/pr-integration.XXXXXX)
+ROOT_WORKSPACE=`pwd`
+CMO_PIPELINES_DIRECTORY=$ROOT_WORKSPACE/cmo-pipelines
+ONCOTREE_DIRECTORY=$ROOT_WORKSPACE/oncotree
+
+ONCOTREE_JAR=$ONCOTREE_DIRECTORY/web/target/oncotree.jar
+IMPORT_SCRIPTS_DIRECTORY=$CMO_PIPELINES_DIRECTORY/import-scripts
+
+JENKINS_USER_HOME_DIRECTORY=/var/lib/jenkins
+JENKINS_PROPERTIES_DIRECTORY=$JENKINS_USER_HOME_DIRECTORY/pipelines-configuration/properties
+APPLICATION_PROPERTIES=application.properties
+
+ONCOTREE_CODE_CONVERTER_TEST_SUCCESS=0
+FAKE_ONCOTREE_VERSION_TEST_SUCCESS=0
+ONCOTREE_CODE_CONVERTER_OUTPUT_TEST_SUCCESS=0
+
+function find_oncotree_process_number {
+    ONCOTREE_PORT_NUMBER=$1
+    ONCOTREE_PROCESS_NUMBER=`netstat -tanp | grep LISTEN | sed 's/\s\s\s*/\t/g' | grep -P ":$ONCOTREE_PORT_NUMBER\t" | cut -f6 | sed 's/\/.*//'`
+    echo $ONCOTREE_PROCESS_NUMBER
+}
+
+function find_free_port { 
+    CHECKED_PORT=10000
+    MAX_PORT=65535
+    EXISTING_PORT=PORTNUMBER
+    # each loop sets EXISTING_PORT to line from netstat (corresponding to CHECKED_PORT)
+    # if grep CHECKED_PORT returns nothing, EXISTING_PORT is unset 
+    # CHECKED_PORT is a free port since not found in netstat output
+    while [[ ! -z $EXISTING_PORT || $CHECKED_PORT -gt $MAX_PORT ]] ; do
+        CHECKED_PORT=$(($CHECKED_PORT + 1))
+        EXISTING_PORT=`netstat -tanp | sed 's/\s\s\s*/\t/g' | grep -P ":$CHECKED_PORT\t"`
+    done
+    if [ $CHECKED_PORT -gt $MAX_PORT ] ; then
+        echo -1
+    else
+        echo $CHECKED_PORT
+    fi
+}
+
+# Copy in ONCOTREE properties and build jar
+rsync $JENKINS_PROPERTIES_DIRECTORY/oncotree/$APPLICATION_PROPERTIES $ONCOTREE_DIRECTORY/web/src/main/resources
+cd $ONCOTREE_DIRECTORY ; mvn package -Dpackaging.type=jar 
+
+#start up ONCOTREE on some port on dashi-dev
+ONCOTREE_PORT=`find_free_port`
+
+TIME_BETWEEN_ONCOTREE_AVAILIBILITY_TESTS=3
+ONCOTREE_DEPLOYMENT_SUCCESS=0
+CURRENT_WAIT_TIME=0
+MAXIMUM_WAIT_TIME=600
+if [ $ONCOTREE_PORT -gt 0 ] ; then
+    java -jar $ONCOTREE_JAR --server.port=$ONCOTREE_PORT &
+
+    # maximum time to wait for oncotree to deploy 2 minutes
+    # every 10 seconds check if job is still running
+    # attempt to hit endpoint - successful return code indicated ONCOTREE has started up
+    ONCOTREE_URL="http://dashi-dev.cbio.mskcc.org:$ONCOTREE_PORT"
+    while [ $ONCOTREE_DEPLOYMENT_SUCCESS -eq 0 ] ; do
+        CURRENT_WAIT_TIME=$(($CURRENT_WAIT_TIME + 10))
+        JOB_RUNNING=`jobs | grep "oncotree.jar" | grep "Running"`
+        if [ -z $JOB_RUNNING ] ; then
+            curl --fail -X GET --header 'Accept: */*' $ONCOTREE_URL       
```

Comment:
I think this might be okay as-is since we just want to know oncotree has started up (and not necessarily the response body). The integration test will check the response body of the endpoints.

This just hits the homepage repeatedly, and will return a 0 exit status (and the html page) when it finishes starting up. Before that it just returns non-zero exit status `curl: (7) couldn't connect to host`